File: pulsar-plugins/pulsar-protocol/src/main/java/fun/platonic/pulsar/protocol/http/Http.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package fun.platonic.pulsar.protocol.http;

import fun.platonic.pulsar.common.config.ImmutableConfig;
import fun.platonic.pulsar.common.proxy.NoProxyException;
import fun.platonic.pulsar.crawl.protocol.Protocol;
import fun.platonic.pulsar.crawl.protocol.ProtocolException;
import fun.platonic.pulsar.crawl.protocol.Response;
import fun.platonic.pulsar.crawl.protocol.http.AbstractHttpProtocol;
import fun.platonic.pulsar.persist.WebPage;

import java.io.IOException;
import java.net.URL;

public class Http extends AbstractHttpProtocol {

    public Http() {
    }

    public Http(ImmutableConfig conf) {
        setConf(conf);
    }

    @Override
    protected Response getResponse(String url, WebPage page, boolean redirect)
            throws ProtocolException, IOException, NoProxyException {
        Response r = null;
        try {
            r = new HttpResponse(this, new URL(url), page);
        } catch (InterruptedException e) {
            Protocol.LOG.error(e.toString());
        }
        return r;
    }
}

File: pulsar-plugins/pulsar-protocol/src/main/kotlin/ai/platon/pulsar/protocol/http/Http.kt
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ai.platon.pulsar.protocol.http

import ai.platon.pulsar.common.config.ImmutableConfig
import ai.platon.pulsar.common.proxy.ProxyException
import ai.platon.pulsar.crawl.protocol.ProtocolException
import ai.platon.pulsar.crawl.protocol.Response
import ai.platon.pulsar.crawl.protocol.http.AbstractNativeHttpProtocol
import ai.platon.pulsar.persist.WebPage
import ai.platon.pulsar.protocol.http.HttpResponse
import java.io.IOException
import java.net.URL

class Http: AbstractNativeHttpProtocol {

    constructor()

    constructor(conf: ImmutableConfig) {
        setConf(conf)
    }

    @Throws(ProtocolException::class, IOException::class, ProxyException::class)
    override fun getResponse(url: String, page: WebPage, redirect: Boolean): Response {
        return HttpResponse(this, URL(url), page)
    }
}
